editor/src/messages/tool/tool_messages/select_tool.rs

```diff
@@ -334,28 +400,81 @@ impl Fsm for SelectToolFsmState {
 					// Check the last (top most) intersection layer.
 					if let Some(intersect_layer_path) = document.document_legacy.intersects_quad_root(quad, render_data).last() {
 						if let Ok(intersect) = document.document_legacy.layer(intersect_layer_path) {
-							match intersect.data {
-								LayerDataType::Text(_) => {
-									responses.push_front(ToolMessage::ActivateTool { tool_type: ToolType::Text }.into());
-									responses.push_back(TextToolMessage::Interact.into());
+							// Shallowest manipulation
```

Comment:
Perhaps the shallowest and deepest manipulation could be extracted into functions to reduce the size of this match arm (and amount of indentation)?